core/src/main/java/com/netflix/iceberg/BaseTableScan.java

```diff
@@ -208,8 +211,8 @@ public InclusiveManifestEvaluator load(Integer specId) {
         TableProperties.SPLIT_LOOKBACK, TableProperties.SPLIT_LOOKBACK_DEFAULT);
 
     return CloseableIterable.transform(
-        CloseableIterable.wrap(planFiles(), files ->
-            new BinPacking.PackingIterable<>(files, splitSize, lookback, FileScanTask::length)),
+        CloseableIterable.wrap(planSplits(splitSize), splits ->
+            new BinPacking.PackingIterable<>(splits, splitSize, lookback, FileScanTask::length)),
```

Comment:
The approach here looks good: take the output of `planFiles`, split files to produce a new iterable, and then pack those splits.